dvc/scm/tree.py

```diff
@@ -60,21 +61,21 @@ def isfile(self, path):
         """Test whether a path is a regular file"""
         return os.path.isfile(path)
 
-    def walk(self, top, topdown=True, ignore_file_handler=None):
+    def walk(self, top, topdown=True, dvcignore=None):
         """Directory tree generator.
 
         See `os.walk` for the docs. Differences:
         - no support for symlinks
         - it could raise exceptions, there is no onerror argument
         """
 
+        if not dvcignore:
+            raise NoDvcIgnoreProvidedError()
```

Comment:
This should not be some dvc error, those are created to show messages to users, this is function signature violation. Use an `assert` or a `TypeError` instead.